crates/cext/src/sparse_observable.rs

```diff
@@ -14,7 +14,11 @@ use std::ffi::{c_char, CString};
 
 use crate::exit_codes::{CInputError, ExitCode};
 use num_complex::Complex64;
-use qiskit_accelerate::sparse_observable::{BitTerm, SparseObservable, SparseTermView};
+use pyo3::ffi::PyObject;
+use pyo3::{Py, Python};
+use qiskit_accelerate::sparse_observable::{
+    BitTerm, PySparseObservable, SparseObservable, SparseTermView,
+};
```

Comment:
All the `Py` stuff here is only used in the `cfg("python_binding")` paths, but we don't see a lint failure because we only run the entire workspace lint, which forcibly enables that feature via `pyext`.  Can we protect the imports for now, so `cargo build -p qiskit-cext` passes build and lint alone, and maybe think of a story for checking that going forwards?

Reply:
You're right this should have the cfg around the python bits. But I think we do have coverage for `qiskit-cext` standalone builds via the c tests already we shouldn't be building via pyext there or doing a whole workspace build.

Reply:
The action workflow uses the `Makefile`, which was building the entire workspace before now.  One of the changes I made to it was to only build the necessary components - I hadn't noticed before when I was looking at the file that it didn't do that.  Fixing it is how I noticed these unused imports - I get the warnings when I build now!

(Btw, in the future we should have the Makefile be configurable so you can choose to build the Python interface using the `Makefile`, because at the moment it just builds `qiskit-cext` with its default features.)

Reply:
Added in: https://github.com/Qiskit/qiskit/pull/13934/commits/547904c244229d1597e34a81334e8012da11a281